### ISSUE TITLE
RDS enabled_cloudwatch_logs_exports log groups not destroyed

### ISSUE THREAD

**Issue by tsembhi** [terraform-providers/terraform-provider-aws#5348]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
0.11.7

### Affected Resource(s)

* aws_db_instance
* aws_cloudwatch_log_group

### Terraform Configuration Files

```hcl
resource "aws_db_instance" "default" {
..
..
 enabled_cloudwatch_logs_exports  = ["audit", "error", "general", "slowquery"]
}

resource "aws_cloudwatch_log_group" "error" {
  name = "/aws/rds/instance/${aws_db_instance.default.identifier}/error"
  retention_in_days = "5"
}
```

### Expected Behavior
Db instance created with export to cloudwatch enabled and  log groups that have a defined expiry (e.g. 5 days). 

### Actual Behavior
The RDS instance seem to create the log groups with a default of "Never Expire". When terraform then gets to the log groups it fails as it cannot create an existing resource. 

```hcl
* aws_cloudwatch_log_group.error: 1 error(s) occurred:
 
* aws_cloudwatch_log_group.error: Creating CloudWatch Log Group failed: ResourceAlreadyExistsException: The specified log group already exists
    status code: 400, request id: 1a81f5cb-8e5c-11e8-903d-553913852fc2:  The CloudWatch Log Group '/aws/rds/instance/prd298-ice1-ice-sql-tal-test/error' already exists.
* 
```

### Steps to Reproduce

Apply a terrform configuration to create RDS MySQL instance with enabled_cloudwatch_logs_exports and attempt to set an expiry on the log groups.

1. `terraform apply`

### Important Factoids
N/A

### More Details
As the log groups have not been created with terraform, it also cannot destroy them.

I have tried the following
* Creating the log groups before the database instance.
 - However when destroying - the log groups get automatically re-created by the RDS instance, and must then be destroyed manually.

The only work around we have now is this:
    1. Apply the terraform config with "enabled_cloudwatch_logs_exports" commented out
    this will create the log groups with an expiry, but not export any logs to them
    2. Re-apply the terraform config with "enabled_cloudwatch_logs_exports" uncommented, and the logs you want listed
    this will enable cloudwatch exporting of the logs and retain the expiry
    3. To destroy - you must first disable "enabled_cloudwatch_logs_exports" ie. commnent out in the terraform config and run an apply -  else the log groups will automatically get re-created by the RDS instance and will need to be manually destroyed.
    4. Then run destroy to destroy the entire stack (including the log groups)

There should be a way to enable cloud watch logs in RDS with an expiry, in a way that can be managed by Terraform.

**Comment by bflad** [comment]
Hi @tsembhi 👋  

By default, [RDS creates a `AWSServiceRoleForRDS` service-linked IAM role that controls what access RDS has to other AWS services](https://aws.amazon.com/blogs/database/monitor-amazon-rds-for-mysql-and-mariadb-logs-with-amazon-cloudwatch/). That service-linked role automatically attaches an IAM policy called `AmazonRDSServiceRolePolicy` which has the following statement:

```
        {
            "Effect": "Allow",
            "Action": [
                "logs:CreateLogGroup"
            ],
            "Resource": [
                "arn:aws:logs:*:*:log-group:/aws/rds/*"
            ]
        },
```

While you cannot detach or edit the AWS managed policy, due to how [IAM policy evaluation](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_evaluation-logic.html) works, theoretically you could add another IAM policy with an explicit `Deny` statement to override RDS's ability to automatically create (unmanaged) CloudWatch Log Groups.

Unfortunately in my brief testing, it does not seem like its possible to customize the RDS IAM service-linked role to control its access better. You may want to reach out to AWS support for help with this particular case unless someone else from the community has more insight or maybe has this figured out.

Basically, if you can prevent RDS from having the above IAM permission, Terraform can fully control creating and deleting the Log Group using the `aws_cloudwatch_log_group` resource.

**Comment by tsembhi** [comment]
Thanks @bflad for the insight :)
I shall have a play with this, and provide any feedback on this issue.

**Comment by tsembhi** [comment]
Sorry for the delay, here is the reponse from AWS:

- You are trying to create a RDS instance using Terraform. 
- For the RDS instance you are enabling the feature 'pushing logs to CloudWatch'. 
- RDS by default creates the log group with NO retention period. 
- Your requirement is to have the retention period of the CloudWatch log to be set to a defined value say 5 days. 

Some of the facts to be considered in this situation are 
- You cannot modify the IAM policies attached to the IAM service role 'AWSServiceRoleForRDS'. 
- RDS uses the service role 'AWSServiceRoleForRDS' for the feature 'pushing logs to CloudWatch'. 
- RDS creates the log group and log stream when you enable this feature while creating or modifying the instance. 

Since RDS tries to create the log group and log stream while creating the instance, I ran a simple test case and below are the sequence of operations. I used CLI to run this test 
1. Created a log group with name "/aws/rds/instance/case-5202863381-mysql/error" using 'aws logs create-log-group' 
2. Created a retention policy on this log group retaining events for 5 days using 'aws logs put-retention-policy' 
3. Created a log stream with name "case-5202863381-mysql" in the above log group using 'aws logs create-log-stream' 
4. Created a DB instance named "case-5202863381-mysql" and this instance is configured to push the 'error' logs to CloudWatch. 

Since the log group and log stream are created prior to the instance creation, the same will be used while pushing the error log entries to CloudWatch. Based on this test case, I can this of the below two work around solutions. 

1. Following the approach mentioned above, create the log group and set the retention period before creating the RDS instance. 
2. Create the RDS instance and once the log group is created by RDS, modify the retention period for the same.
